src/git.rs

```diff
@@ -39,26 +40,33 @@ fn lookup_rev<'rev>(repo: &'rev Repository, rev: &str) -> Result<Git2Commit<'rev
 }
 
 fn get_repo() -> Result<Repository, Error> {
-    let loc = Path::new("rust.git");
-    match (RUST_SRC_REPO, loc.exists()) {
-        (Some(_), _) | (_, true) => {
-            let path = RUST_SRC_REPO.map(Path::new).unwrap_or(loc);
-            eprintln!("opening existing repository at {:?}", path);
-            let repo = Repository::open(path)?;
-            {
-                eprintln!("refreshing repository");
-                let mut remote = repo
-                    .find_remote("origin")
-                    .or_else(|_| repo.remote_anonymous("origin"))?;
-                remote.fetch(&["master"], None, None)?;
-            }
-            Ok(repo)
+    fn open(repo: &Path) -> Result<Repository, Error> {
+        eprintln!("opening existing repository at {:?}", repo);
+        let repo = Repository::open(repo)?;
+        {
+            eprintln!("refreshing repository");
+            // Sometimes origin point to github fork of contributor,
+            // let's inspect upstream remote first.
+            let mut remote = repo
+                .find_remote("upstream")
```

Comment:
I understand what you're trying to do here but wouldn't it be better to pass this also as an env var?.

Reply:
I am think about using `remote_anonymous` with rustc URL instead.
This has the advantages that we don't have to guess the remote names the and their URLs.
Another problem is sometime the remote URL is alias for ssh host, does git2/we handle it.